### ISSUE TITLE
Coverage is Always Zero

### ISSUE THREAD

**Issue by JimLynchCodes** [karma-runner/karma#3445]
I have been trying to add coverage reporting for unit testing of a fresh Angular project ***for weeks*** with no success...

Would be great if someone could help me. 🙏

My karma.conf.js:
```
// Karma configuration file, see link for more information
// https://karma-runner.github.io/1.0/config/configuration-file.html

module.exports = function (config) {
  config.set({
    basePath: '',
    frameworks: ['jasmine', '@angular-devkit/build-angular'],
    plugins: [
      require('karma-jasmine'),
      require('karma-coverage'),
      require('karma-chrome-launcher'),
      require('karma-jasmine-html-reporter'),
      require('karma-coverage-istanbul-reporter'),
      require('@angular-devkit/build-angular/plugins/karma')
    ],
    client: {
      clearContext: false // leave Jasmine Spec Runner output visible in browser
    },
    coverageIstanbulReporter: {
      dir: require('path').join(__dirname, './coverage/TddBdd2020'),
      reports: ['html', 'lcovonly', 'text-summary'],
      fixWebpackSourcePaths: true
    },
    // list of files / patterns to load in the browser
    files: [
      '**/src/**/*.js',
      // 'test/spec/**/*.ts'
    ],
    exclude: ['**/node_modules'],
    preprocessors: {'**src/**/*.ts': ['coverage']},
    reporters: ['coverage-istanbul'],
    port: 9876,
    colors: true,
    logLevel: config.LOG_INFO,
    autoWatch: true,
    browsers: ['Chrome'],
    singleRun: false,
    restartOnFileChange: true,
    verbose: true
  });
};
```

output: 

![Screen Shot 2020-03-13 at 10 42 48 PM](https://user-images.githubusercontent.com/5354163/76673941-ab288480-6580-11ea-8737-0e8f4a8a0402.png)

**Comment by JimLynchCodes** [comment]
bump! 🙃

**Comment by johnjbarton** [comment]
> preprocessors: {'**src/**/*.ts': ['coverage']},

Does that work? I would be surprised since this is a JS tool.

> logLevel: config.LOG_INFO,
Try LOG_DEBUG and look through the log. It will report coverage events.

**Comment by JimLynchCodes** [comment]
I am trying to use it on an Angular project where everything is in TypeScript.

I tried with `ts` and with `js`, but the coverage report never finds any files...

**Comment by JimLynchCodes** [comment]
I set the log level to LOG_DEBUG, and it prints out a lot of text to the console. However, I don't see anything in there that says what's wrong with the coverage...

```
TddBdd2020 (master) 🍓 npm test

> tdd-bdd2020@0.0.0 test /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020
> ng test

18 03 2020 20:13:11.339:DEBUG [config]: Loading config /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/karma.conf.js
18 03 2020 20:13:11.345:DEBUG [karma-server]: Final config [{"LOG_DISABLE":"1","LOG_ERROR":"2","LOG_WARN":"3","LOG_INFO":"4","LOG_DEBUG":"5","frameworks":"6","protocol":"7","port":9876,"listenAddress":"8","hostname":"9","httpsServerConfig":"10","basePath":"11","files":"12","browserConsoleLogOptions":"13","customContextFile":null,"customDebugFile":null,"customClientContextFile":null,"exclude":"14","logLevel":"5","colors":true,"autoWatch":true,"autoWatchBatchDelay":250,"restartOnFileChange":true,"usePolling":false,"reporters":"15","singleRun":false,"browsers":"16","captureTimeout":60000,"proxies":"17","proxyValidateSSL":true,"preprocessors":"18","urlRoot":"19","reportSlowerThan":0,"loggers":"20","transports":"21","forceJSONP":false,"plugins":"22","client":"23","defaultClient":"23","browserDisconnectTimeout":2000,"browserDisconnectTolerance":0,"browserNoActivityTimeout":30000,"processKillTimeout":2000,"concurrency":null,"failOnEmptyTestSuite":true,"retryLimit":2,"detached":false,"crossOriginAttribute":true,"browserSocketTimeout":20000,"configFile":"24","coverageIstanbulReporter":"25"},"OFF","ERROR","WARN","INFO","DEBUG",["26","27"],"http:","0.0.0.0","localhost",{},"/Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020",["28","29"],{"level":"30","format":"31","terminal":true},["24"],["32","33","34"],["35"],{},{},"/",["36"],["37","38"],["39","40","41","42","43"],{"args":"44","useIframe":true,"runInParent":false,"captureConsole":true,"clearContext":false},"/Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/karma.conf.js",{"dir":"45","reports":"46","fixWebpackSourcePaths":true},"jasmine","@angular-devkit/build-angular",{"pattern":"47","served":true,"included":true,"watched":true,"nocache":false,"weight":"48"},{"pattern":"49","served":true,"included":true,"watched":true,"nocache":false,"weight":"50"},"debug","%b %T: %m","progress","kjhtml","coverage-istanbul","Chrome",{"type":"51","layout":"52"},"polling","websocket",{"framework:jasmine":"53"},{"launcher:Chrome":"54","launcher:ChromeHeadless":"55","launcher:ChromeCanary":"56","launcher:ChromeCanaryHeadless":"57","launcher:Chromium":"58","launcher:ChromiumHeadless":"59","launcher:Dartium":"60","test":"61"},{"reporter:kjhtml":"62"},{"reporter:coverage-istanbul":"63"},{"framework:@angular-devkit/build-angular":"64","reporter:@angular-devkit/build-angular--sourcemap-reporter":"65","reporter:@angular-devkit/build-angular--event-reporter":"66","middleware:@angular-devkit/build-angular--blocker":"67","middleware:@angular-devkit/build-angular--fallback":"68"},[],"/Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/coverage/TddBdd2020",["69","70","71"],"/Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/app/js/**/*.ts",[1,1,1,0,0,0],"/Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/test/spec/**/*.ts",[1,1,1,0,0,0],"console",{"type":"72","pattern":"73"},["74",null],["75",null],["75",null],["75",null],["75",null],["75",null],["75",null],["75",null],{},["75",null],["75",null],["74",null],["75",null],["75",null],["74",null],["74",null],"html","lcovonly","text-summary","pattern","%[%d{DATE}:%p [%c]: %]%m","factory","type"]
18 03 2020 20:13:11.346:DEBUG [plugin]: Loading inlined plugin (defining framework:jasmine).
18 03 2020 20:13:11.346:DEBUG [plugin]: Loading inlined plugin (defining launcher:Chrome, launcher:ChromeHeadless, launcher:ChromeCanary, launcher:ChromeCanaryHeadless, launcher:Chromium, launcher:ChromiumHeadless, launcher:Dartium, test).
18 03 2020 20:13:11.346:DEBUG [plugin]: Loading inlined plugin (defining reporter:kjhtml).
18 03 2020 20:13:11.346:DEBUG [plugin]: Loading inlined plugin (defining reporter:coverage-istanbul).
18 03 2020 20:13:11.346:DEBUG [plugin]: Loading inlined plugin (defining framework:@angular-devkit/build-angular, reporter:@angular-devkit/build-angular--sourcemap-reporter, reporter:@angular-devkit/build-angular--event-reporter, middleware:@angular-devkit/build-angular--blocker, middleware:@angular-devkit/build-angular--fallback).
10% building 2/2 modules 0 active18 03 2020 20:13:12.073:DEBUG [web-server]: Instantiating middleware
18 03 2020 20:13:12.086:DEBUG [reporter]: Trying to load reporter: kjhtml
18 03 2020 20:13:12.087:DEBUG [reporter]: Trying to load color-version of reporter: kjhtml (kjhtml_color)
18 03 2020 20:13:12.087:DEBUG [reporter]: Couldn't load color-version.
18 03 2020 20:13:12.087:DEBUG [reporter]: Trying to load reporter: coverage-istanbul
18 03 2020 20:13:12.087:DEBUG [reporter]: Trying to load color-version of reporter: coverage-istanbul (coverage-istanbul_color)
18 03 2020 20:13:12.087:DEBUG [reporter]: Couldn't load color-version.
18 03 2020 20:13:12.087:DEBUG [reporter]: Trying to load reporter: @angular-devkit/build-angular--sourcemap-reporter
18 03 2020 20:13:12.087:DEBUG [reporter]: Trying to load color-version of reporter: @angular-devkit/build-angular--sourcemap-reporter (@angular-devkit/build-angular--sourcemap-reporter_color)
18 03 2020 20:13:12.087:DEBUG [reporter]: Couldn't load color-version.
18 03 2020 20:13:12.087:DEBUG [reporter]: Trying to load reporter: @angular-devkit/build-angular--event-reporter
18 03 2020 20:13:12.088:DEBUG [reporter]: Trying to load color-version of reporter: @angular-devkit/build-angular--event-reporter (@angular-devkit/build-angular--event-reporter_color)
18 03 2020 20:13:12.088:DEBUG [reporter]: Couldn't load color-version.
18 03 2020 20:13:13.344:WARN [filelist]: Pattern "/Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/app/js/**/*.ts" does not match any file.
18 03 2020 20:13:13.344:WARN [filelist]: Pattern "/Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/test/spec/**/*.ts" does not match any file.
18 03 2020 20:13:13.352:DEBUG [karma-server]: List of files has changed, trying to execute
18 03 2020 20:13:13.353:WARN [karma]: No captured browser, open http://localhost:9876/
18 03 2020 20:13:13.425:DEBUG [watcher]: Watching "/Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/app/js"
18 03 2020 20:13:13.425:DEBUG [watcher]: Watching "/Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/test/spec"
18 03 2020 20:13:13.425:INFO [karma-server]: Karma v4.1.0 server started at http://0.0.0.0:9876/
18 03 2020 20:13:13.426:INFO [launcher]: Launching browsers Chrome with concurrency unlimited
18 03 2020 20:13:13.437:INFO [launcher]: Starting browser Chrome
18 03 2020 20:13:13.438:DEBUG [launcher]: null -> BEING_CAPTURED
18 03 2020 20:13:13.438:DEBUG [temp-dir]: Creating temp dir at /var/folders/jk/1yhv_4k52pg2y98xs_ct0dh80000gn/T/karma-22387451
18 03 2020 20:13:13.438:DEBUG [launcher]: /Applications/Google Chrome.app/Contents/MacOS/Google Chrome --user-data-dir=/var/folders/jk/1yhv_4k52pg2y98xs_ct0dh80000gn/T/karma-22387451 --no-default-browser-check --no-first-run --disable-default-apps --disable-popup-blocking --disable-translate --disable-background-timer-throttling --disable-renderer-backgrounding --disable-device-discovery-notifications http://localhost:9876/?id=22387451
18 03 2020 20:13:16.129:WARN [filelist]: Pattern "/Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/app/js/**/*.ts" does not match any file.
18 03 2020 20:13:16.129:WARN [filelist]: Pattern "/Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/test/spec/**/*.ts" does not match any file.
18 03 2020 20:13:16.130:DEBUG [web-server]: serving: /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/karma/static/client.html
18 03 2020 20:13:16.139:DEBUG [karma-server]: List of files has changed, trying to execute
18 03 2020 20:13:16.140:WARN [karma]: No captured browser, open http://localhost:9876/
18 03 2020 20:13:16.147:DEBUG [web-server]: serving: /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/karma/static/karma.js
18 03 2020 20:13:16.211:DEBUG [karma-server]: A browser has connected on socket am_GVUovkZbTlaWWAAAA
18 03 2020 20:13:16.212:DEBUG [web-server]: upgrade /socket.io/?EIO=3&transport=websocket&sid=am_GVUovkZbTlaWWAAAA
18 03 2020 20:13:16.222:DEBUG [Chrome 80.0.3987 (Mac OS X 10.15.3)]: undefined -> CONNECTED
18 03 2020 20:13:16.222:INFO [Chrome 80.0.3987 (Mac OS X 10.15.3)]: Connected on socket am_GVUovkZbTlaWWAAAA with id 22387451
18 03 2020 20:13:16.222:DEBUG [launcher]: BEING_CAPTURED -> CAPTURED
18 03 2020 20:13:16.222:DEBUG [launcher]: Chrome (id 22387451) captured in 2.796 secs
18 03 2020 20:13:16.222:DEBUG [karma]: All browsers are ready, executing
18 03 2020 20:13:16.222:DEBUG [karma]: Captured 1 browsers
18 03 2020 20:13:16.225:DEBUG [web-server]: serving: /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/karma/static/favicon.ico
18 03 2020 20:13:16.227:DEBUG [middleware:karma]: custom files /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/@angular-devkit/build-angular/src/angular-cli-files/plugins/karma-context.html /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/@angular-devkit/build-angular/src/angular-cli-files/plugins/karma-debug.html null
18 03 2020 20:13:16.228:DEBUG [middleware:karma]: Serving customContextFile /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/@angular-devkit/build-angular/src/angular-cli-files/plugins/karma-context.html
18 03 2020 20:13:16.228:DEBUG [web-server]: serving: /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/@angular-devkit/build-angular/src/angular-cli-files/plugins/karma-context.html
18 03 2020 20:13:16.234:DEBUG [middleware:source-files]: Requesting /_karma_webpack_/runtime.js
18 03 2020 20:13:16.234:DEBUG [middleware:source-files]: Fetching /_karma_webpack_/runtime.js
18 03 2020 20:13:16.235:DEBUG [middleware:source-files]: Requesting /_karma_webpack_/polyfills.js
18 03 2020 20:13:16.235:DEBUG [middleware:source-files]: Fetching /_karma_webpack_/polyfills.js
18 03 2020 20:13:16.236:DEBUG [web-server]: serving: /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/karma/static/context.js
18 03 2020 20:13:16.236:DEBUG [middleware:source-files]: Requesting /base/node_modules/source-map-support/browser-source-map-support.js?0a776857d457dcbb7a4bc82b263431477788990d
18 03 2020 20:13:16.236:DEBUG [middleware:source-files]: Fetching /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/source-map-support/browser-source-map-support.js
18 03 2020 20:13:16.237:DEBUG [web-server]: serving (cached): /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/source-map-support/browser-source-map-support.js
18 03 2020 20:13:16.237:DEBUG [middleware:source-files]: Requesting /base/node_modules/karma-source-map-support/lib/client.js?fcb459e84379f1ce8e91fe200d87463f3780ca1b
18 03 2020 20:13:16.237:DEBUG [middleware:source-files]: Fetching /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/karma-source-map-support/lib/client.js
18 03 2020 20:13:16.237:DEBUG [middleware:source-files]: Requesting /base/node_modules/jasmine-core/lib/jasmine-core/jasmine.js?be3ff9a5e2d6d748de5b900ac3c6d9603e2942a7
18 03 2020 20:13:16.237:DEBUG [middleware:source-files]: Fetching /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/jasmine-core/lib/jasmine-core/jasmine.js
18 03 2020 20:13:16.238:DEBUG [web-server]: serving (cached): /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/karma-source-map-support/lib/client.js
18 03 2020 20:13:16.238:DEBUG [web-server]: serving (cached): /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/jasmine-core/lib/jasmine-core/jasmine.js
18 03 2020 20:13:16.238:DEBUG [middleware:source-files]: Requesting /base/node_modules/karma-jasmine-html-reporter/src/css/jasmine.css?2a656d04accedfa255285653692ead769b02beed
18 03 2020 20:13:16.238:DEBUG [middleware:source-files]: Fetching /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/karma-jasmine-html-reporter/src/css/jasmine.css
18 03 2020 20:13:16.239:DEBUG [web-server]: serving (cached): /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/karma-jasmine-html-reporter/src/css/jasmine.css
18 03 2020 20:13:16.239:DEBUG [middleware:source-files]: Requesting /base/node_modules/karma-jasmine/lib/boot.js?945a38bf4e45ad2770eb94868231905a04a0bd3e
18 03 2020 20:13:16.239:DEBUG [middleware:source-files]: Fetching /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/karma-jasmine/lib/boot.js
18 03 2020 20:13:16.242:DEBUG [web-server]: serving (cached): /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/karma-jasmine/lib/boot.js
18 03 2020 20:13:16.242:DEBUG [middleware:source-files]: Requesting /base/node_modules/karma-jasmine/lib/adapter.js?3098011cfe00faa2a869a8cffce13f3befc1a035
18 03 2020 20:13:16.242:DEBUG [middleware:source-files]: Fetching /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/karma-jasmine/lib/adapter.js
18 03 2020 20:13:16.242:DEBUG [middleware:source-files]: Requesting /base/node_modules/karma-jasmine-html-reporter/src/lib/html.jasmine.reporter.js?16e2680e2d266f1753833c5ae8def4cc08cec58c
18 03 2020 20:13:16.242:DEBUG [middleware:source-files]: Fetching /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/karma-jasmine-html-reporter/src/lib/html.jasmine.reporter.js
18 03 2020 20:13:16.242:DEBUG [middleware:source-files]: Requesting /base/node_modules/karma-jasmine-html-reporter/src/lib/adapter.js?a5e54e132e216edf081451bd2247db26b5e5b082
18 03 2020 20:13:16.242:DEBUG [middleware:source-files]: Fetching /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/karma-jasmine-html-reporter/src/lib/adapter.js
18 03 2020 20:13:16.243:DEBUG [middleware:source-files]: Requesting /_karma_webpack_/styles.js
18 03 2020 20:13:16.243:DEBUG [middleware:source-files]: Fetching /_karma_webpack_/styles.js
18 03 2020 20:13:16.243:DEBUG [web-server]: serving (cached): /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/karma-jasmine/lib/adapter.js
18 03 2020 20:13:16.243:DEBUG [web-server]: serving (cached): /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/karma-jasmine-html-reporter/src/lib/html.jasmine.reporter.js
18 03 2020 20:13:16.243:DEBUG [web-server]: serving (cached): /Users/jim/Git-Projects/Angular-TDD-BDD-2020/TddBdd2020/node_modules/karma-jasmine-html-reporter/src/lib/adapter.js
18 03 2020 20:13:16.243:DEBUG [middleware:source-files]: Requesting /_karma_webpack_/scripts.js
18 03 2020 20:13:16.243:DEBUG [middleware:source-files]: Fetching /_karma_webpack_/scripts.js
18 03 2020 20:13:16.244:DEBUG [middleware:source-files]: Requesting /_karma_webpack_/vendor.js
18 03 2020 20:13:16.244:DEBUG [middleware:source-files]: Fetching /_karma_webpack_/vendor.js
18 03 2020 20:13:16.245:DEBUG [middleware:source-files]: Requesting /_karma_webpack_/main.js
18 03 2020 20:13:16.245:DEBUG [middleware:source-files]: Fetching /_karma_webpack_/main.js
18 03 2020 20:13:16.441:DEBUG [middleware:source-files]: Requesting /_karma_webpack_/vendor.js
18 03 2020 20:13:16.442:DEBUG [middleware:source-files]: Fetching /_karma_webpack_/vendor.js
18 03 2020 20:13:16.526:DEBUG [middleware:source-files]: Requesting /_karma_webpack_/vendor.js
18 03 2020 20:13:16.526:DEBUG [middleware:source-files]: Fetching /_karma_webpack_/vendor.js
18 03 2020 20:13:17.513:DEBUG [middleware:source-files]: Requesting /_karma_webpack_/main.js
18 03 2020 20:13:17.513:DEBUG [middleware:source-files]: Fetching /_karma_webpack_/main.js
18 03 2020 20:13:17.516:DEBUG [middleware:source-files]: Requesting /_karma_webpack_/main.js
18 03 2020 20:13:17.516:DEBUG [middleware:source-files]: Fetching /_karma_webpack_/main.js
18 03 2020 20:13:17.580:DEBUG [Chrome 80.0.3987 (Mac OS X 10.15.3)]: CONNECTED -> EXECUTING
Chrome 80.0.3987 (Mac OS X 10.15.3): Executed 3 of 3 SUCCESS (0 secs / 0.238 secs)
Chrome 80.0.3987 (Mac OS X 10.15.3): Executed 3 of 3 SUCCESS (0.284 secs / 0.238 secs)
TOTAL: 3 SUCCESS
TOTAL: 3 SUCCESS
TOTAL: 3 SUCCESS
18 03 2020 20:13:17.872:DEBUG [reporter.coverage-istanbul]: Writing coverage reports: [ 'html', 'lcovonly', 'text-summary' ]

=============================== Coverage summary ===============================
Statements   : Unknown% ( 0/0 )
Branches     : Unknown% ( 0/0 )
Functions    : Unknown% ( 0/0 )
Lines        : Unknown% ( 0/0 )
================================================================================
```

**Comment by johnjbarton** [comment]
https://www.npmjs.com/package/karma-coverage-istanbul-reporter
"This is a reporter only and does not perform the actual instrumentation of
your code. Babel users should use the istanbul babel plugin
<https://github.com/istanbuljs/babel-plugin-istanbul> to instrument your
code and webpack + typescript users should use the
istanbul-instrumenter-loader
<https://github.com/deepsweet/istanbul-instrumenter-loader> and then use
this karma reporter to do the actual reporting. See the test config
<https://github.com/mattlewis92/karma-coverage-istanbul-reporter/blob/master/test/karma.conf.js>
for
an e2e example of how to combine them."

**Comment by JimLynchCodes** [comment]
Thanks so much @johnjbarton 🙏

the thing is... I am already using `karma-coverage-istanbul-reporter`, and it doesn't seem to be doing anything.

Thanks for your example config file. The thing is, my karma config from a recently scaffolded Angular project using `ng new` looks _WAY_ different from the one you provided so I am still extremely confused as to what I'm supposed to change. 

Here is my `karma.conf.js`: https://github.com/JimLynchCodes/Angular-TDD-BDD-2020/blob/master/TddBdd2020/karma.conf.js

Can you please instruct me on what exactly I need to change here to get it working? 🤔

Thanks! ❤️

**Comment by johnjbarton** [comment]
> the thing is... I am already using karma-coverage-istanbul-reporter, and it doesn't seem to be doing anything.

That is the expected result. That reporter just reports coverage in a nicer way. It does not compute coverage. Thus you get zero coverage, reported nicely.

**Comment by JimLynchCodes** [comment]
@johnjbarton So you are saying there is no way to get the coverage to report on the files...? 🤔

**Comment by johnjbarton** [comment]
No, I am saying that you need to add something like karma-coverage to your config.

**Comment by JimLynchCodes** [comment]
For anyone else having this problem, I finally figured it out after finding [this blog post](https://www.codewithjason.com/add-test-coverage-report-angular-cli-project/?unapproved=7124&moderation-hash=7bcf8bff3bc7a8b573288d7b20f1a847#comment-7124).

In the end, I was able to use the default karma config file from a fresh `ng new` scaffold with no changes to the karma file at all.

Just install the dependencies:
```
npm install karma karma-jasmine karma-chrome-launcher karma-jasmine-html-reporter karma-coverage-istanbul-reporter
```

And then add the `--code-coverage` flag when running the tests:
```
ng test --code-coverage
```

And now finally, it sees my files. 🎉

![Screen Shot 2020-03-19 at 10 12 26 PM](https://user-images.githubusercontent.com/5354163/77130548-c5ee7380-6a2e-11ea-835f-9b94ac5b5c03.png)
